src/marqo/core/unstructured_vespa_index/unstructured_document.py

```diff
@@ -0,0 +1,180 @@
+import json
+from copy import deepcopy
+from typing import List, Dict, Any
+
+from pydantic import Field, BaseModel
+
+from marqo.core import constants as index_constants
+from marqo.core.exceptions import VespaDocumentParsingError
+from marqo.core.unstructured_vespa_index import common as unstructured_common
+from marqo.core.unstructured_vespa_index import constants as unstructured_constants
+
+
+class UnstructuredVespaDocumentFields(BaseModel):
```

Comment:
All `ValueErrors` that are raised during parsing should be `VespaDocumentParsingError` instead